lib/client/interfaces.go

```diff
@@ -84,6 +81,70 @@ func NewKey() (key *Key, err error) {
 	}, nil
 }
 
+// KeyFromIdentityFile loads the private key + certificate
+// from an identity file into a Key.
+func KeyFromIdentityFile(path string) (*Key, error) {
```

Comment:
This function has quite an overlap with `LoadIdentity` from `tool/tsh/common/identity.go`. However that other function doesn't seem to be referenced at all. Could you perhaps remove that one, and add anything missing from there to here? I already noticed an `ident.Certs.SSH` check is done there that's not present here.

Reply:
This was supposed to replace `LoadIdentity`, thanks for pointing this out. Probably a merge issue at some point. And the Certs check is happening in `ReadIdentityFile` which is still called, but I'll look over the rest.

Reply:
The `ident.Certs.SSH` is still missing...?

Reply:
That check was moved to `ReadIdentityFile`, which is called immediately by `KeyFromIdentityFile`, so the check is still in the call path.